Source/WebCore/Modules/cache/DOMCache.cpp

```diff
@@ -41,15 +41,15 @@
 namespace WebCore {
 using namespace WebCore::DOMCacheEngine;
 
-Ref<DOMCache> DOMCache::create(ScriptExecutionContext& context, String&& name, DOMCacheIdentifier identifier, Ref<CacheStorageConnection>&& connection)
+Ref<DOMCache> DOMCache::create(ScriptExecutionContext* context, String&& name, DOMCacheIdentifier identifier, Ref<CacheStorageConnection>&& connection)
 {
     auto cache = adoptRef(*new DOMCache(context, WTFMove(name), identifier, WTFMove(connection)));
     cache->suspendIfNeeded();
     return cache;
 }
 
-DOMCache::DOMCache(ScriptExecutionContext& context, String&& name, DOMCacheIdentifier identifier, Ref<CacheStorageConnection>&& connection)
-    : ActiveDOMObject(&context)
+DOMCache::DOMCache(ScriptExecutionContext* context, String&& name, DOMCacheIdentifier identifier, Ref<CacheStorageConnection>&& connection)
+    : ActiveDOMObject(context)
```

Comment:
It's surprising that `DOMCache` can exist in the absence of an active `ScriptExecutionContext`, but I see that `ActiveDOMObject` does expect this, so this seems correct.

Reply:
Yes, having a null scriptExecution context is something that is expected by ActiveDOMObject.